### PR TITLE
docs(bottle): Document `failed_request_status_codes`

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
## DESCRIBE YOUR PR

Document Bottle integration's `failed_request_status_codes` option, added in #3618.

## IS YOUR CHANGE URGENT?  

- [X] None: Not urgent, can wait up to 1 week+

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [x] Checked Vercel preview for correctness, including links
- [x] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [x] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

### CODE REVIEW

**Review by sentrivana — APPROVED** [review]

File: docs/platforms/python/integrations/bottle/index.mdx [hunk 1]
```diff
@@ -90,6 +91,22 @@ def myendpoint():
...
 - If you set `transaction_style="url"`, the transaction name is `"/myurl/<foo>"`, since that is the URL path.
 
+
+### `failed_request_status_codes`
+
+A `set` of integers that will determine when an [`HTTPResponse`](https://bottlepy.org/docs/dev/api.html#bottle.HTTPResponse), which is raised or returned by the request handler, should be reported to Sentry. The `HTTPResponse` is reported to Sentry if its status code is contained in the `failed_request_status_codes` set.
+
+Examples of valid values for `failed_request_status_codes`:
+
+- `{500}` will only report `HTTPResponse` with status 500.
```
Comment (coolguyzone) [inline docs/platforms/python/integrations/bottle/index.mdx:101]:
```suggestion
- `{500}` will only report an `HTTPResponse` with a 500 status.
```

File: docs/platforms/python/integrations/bottle/index.mdx [hunk 2]
```diff
@@ -90,6 +91,22 @@ def myendpoint():
...
+- `{400, *range(500, 600)}` will report `HTTPResponse` with status 400 as well as those in the 5xx range.
```
Comment (coolguyzone) [inline docs/platforms/python/integrations/bottle/index.mdx:102]:
```suggestion
- `{400, *range(500, 600)}` will report any `HTTPResponse` with a 400 status as well as those in the 5xx range.
```

File: docs/platforms/python/integrations/bottle/index.mdx [hunk 3]
```diff
@@ -90,6 +91,22 @@ def myendpoint():
...
+- `set()` (the empty set) will not report any `HTTPResponse` to Sentry.
+
+The default is `{*range(500, 600)}`, meaning that any `HTTPResponse` with a status in the 5xx range is reported to Sentry.
+
+Regardless of how `failed_request_status_codes` is set, any exceptions raised by the handler, which are not of type `HTTPResponse` (or a subclass) are reported to Sentry. For example, if your request handler raises an unhandled `AttributeError`, the `AttributeError` gets reported to Sentry, even if you have set `failed_request_status_codes=set()`.
```
Comment (coolguyzone) [inline docs/platforms/python/integrations/bottle/index.mdx:107]:
```suggestion
Regardless of how `failed_request_status_codes` is configured, any exceptions raised by the handler which are not of type `HTTPResponse` (or a subclass) are reported to Sentry. For example, if your request handler raises an unhandled `AttributeError`, the `AttributeError` gets reported to Sentry, even if you have set `failed_request_status_codes=set()`.
```

**Review by coolguyzone — APPROVED** [review]
Thanks for adding! Left a couple of minor suggestions, looks good to me!

File: docs/platforms/python/integrations/bottle/index.mdx (hunk 1 above)
Comment (szokeasaurusrex) [reply docs/platforms/python/integrations/bottle/index.mdx:101]: I would prefer to keep this one (and the next comment) as is. This is in alignment with the language used in the [docs for AIOHTTP's](https://github.com/getsentry/sentry-docs/pull/11433) `failed_request_status_codes` option

File: docs/platforms/python/integrations/bottle/index.mdx (hunk 2 above)
Comment (szokeasaurusrex) [reply docs/platforms/python/integrations/bottle/index.mdx:102]: see response above

File: docs/platforms/python/integrations/bottle/index.mdx (hunk 3 above)
Comment (szokeasaurusrex) [reply docs/platforms/python/integrations/bottle/index.mdx:107]: I agree that "configured" is better here; however, if I remember my English classes correctly 😅, the comma needs to be there. However, I did notice that I am missing a comment after the "(or a subclass)," so I will add this.

Comment (szokeasaurusrex) [reply docs/platforms/python/integrations/bottle/index.mdx:107]: Actually, I think I will rewrite this and just say that any "non-`HTTPResponse`" exceptions are reported

**Pull request merged**